coral/src/app/features/configuration/clusters/components/ClusterConnectHelpModal.tsx

```diff
@@ -0,0 +1,227 @@
+import {
+  Box,
+  Button,
+  Icon,
+  PositionerPlacement,
+  Spacing,
+  Tooltip,
+  Typography,
+} from "@aivenio/aquarium";
+import duplicate from "@aivenio/aquarium/icons/duplicate";
+import link from "@aivenio/aquarium/icons/link";
+import { FormEvent, useEffect, useRef, useState } from "react";
+import { Modal } from "src/app/components/Modal";
+import { ClusterDetails } from "src/domain/cluster";
+
+const copyToClipboard = async (text: string): Promise<void> => {
+  try {
+    await navigator.clipboard.writeText(text);
+  } catch (error) {
+    console.error("Failed to copy to clipboard:", error);
+  }
+};
+
+const ClipBoard = ({
+  text,
+  description,
+}: {
+  text: string;
+  description?: string;
+}) => {
+  const feedbackTimerRef = useRef<number>();
+
+  const [showCopyFeedback, setShowCopyFeedback] = useState(false);
+
+  const clearTimeouts = () => {
+    window.clearTimeout(feedbackTimerRef.current);
+  };
+
+  useEffect(() => () => clearTimeouts(), []);
+
+  function handleCopy(event: FormEvent) {
+    event.preventDefault();
+    copyToClipboard(text);
+    setShowCopyFeedback(true);
+    feedbackTimerRef.current = window.setTimeout(
+      () => setShowCopyFeedback(false),
+      1000
+    );
+  }
+
+  if (showCopyFeedback) {
+    return (
+      <Tooltip
+        key="copied"
+        content="Copied"
+        isOpen
+        placement={PositionerPlacement.top}
+      >
+        <Button.SecondaryGhost
+          key="copy-button"
+          aria-label="Copy"
+          onClick={handleCopy}
+          icon={duplicate}
+        >
+          {description}
+        </Button.SecondaryGhost>
+      </Tooltip>
+    );
+  }
+
+  return (
+    <Tooltip
+      key="copy-to-clipboard"
+      content="Copy to clipboard"
+      placement={PositionerPlacement.top}
+    >
+      <Button.SecondaryGhost
+        key="copy-button"
+        aria-label="Copy"
+        onClick={handleCopy}
+        icon={duplicate}
+      >
+        {description}
+      </Button.SecondaryGhost>
+    </Tooltip>
+  );
+};
+
+const getApplicationPropertiesString = ({
```

Comment:
How about moving this out as a helper / utils function? It would keep the component tidier (is that a word 🤔 ) and make it more readable.

Reply:
I personally think that if a function is exclusively used in  a single component, it belongs in the component file. I view it as slightly excessive atomisation to systematically break off things that are single use in other files :o

Reply:
From the structure we planned in the beginning, this file could be co-located right in the same directory, just in a different file - or in `.../configuration/cluster/utils/file.ts`. So from the location very clearly coupled to this component. 

I think a clear function name with clear input and output types is more than enough information in this file, you would only need to jump to a different file if you need to look into details.  Then the string formatting function can do it's thing while the component function is responsible for rendering the thing in the end. 

This enables easier unit testing for the function (it's only a TS test) and the heavier component tests can focus on the render related tests. This is not AS important for us though, as we don't have slow tests, I just like to optimise from the beginning.

But there's no need to change that, it was only a suggestion! :D I tend to lean more pro separation in these cases, but that's a personal preference in our case.